crates/rattler_repodata_gateway/src/sparse/mod.rs

```diff
@@ -55,50 +55,53 @@ enum SparseRepoDataInner {
 impl SparseRepoDataInner {
     fn borrow_repo_data(&self) -> &LazyRepoData<'_> {
         match self {
-            SparseRepoDataInner::Memmapped(inner) => inner.borrow_repo_data(),
-            SparseRepoDataInner::Bytes(inner) => inner.borrow_repo_data(),
+            #[cfg(any(unix, windows))]
+            SparseRepoDataInner::Memmapped(inner) => inner.borrow_dependent(),
+            SparseRepoDataInner::Bytes(inner) => inner.borrow_dependent(),
         }
     }
 }
 
-/// A struct that holds a memory map of a `repodata.json` file and also a
-/// self-referential field which indexes the data in the memory map with a
-/// sparsely parsed json struct. See [`LazyRepoData`].
-#[ouroboros::self_referencing]
-struct MemmappedSparseRepoDataInner {
-    /// Memory map of the `repodata.json` file
-    memory_map: memmap2::Mmap,
-
-    /// Sparsely parsed json content of the memory map. This data struct holds
-    /// references into the memory map so we have to use ouroboros to make
-    /// this legal.
-    #[borrows(memory_map)]
-    #[covariant]
-    repo_data: LazyRepoData<'this>,
-}
-
-/// A struct that holds a reference to the bytes of a `repodata.json` file and
-/// also a self-referential field which indexes the data in the `bytes` with a
-/// sparsely parsed json struct. See [`LazyRepoData`].
-#[ouroboros::self_referencing]
-struct BytesSparseRepoDataInner {
-    /// Bytes of the `repodata.json` file
-    bytes: Bytes,
-
-    /// Sparsely parsed json content of the file's bytes. This data struct holds
-    /// references into the bytes so we have to use ouroboros to make this
-    /// legal.
-    #[borrows(bytes)]
-    #[covariant]
-    repo_data: LazyRepoData<'this>,
-}
+// A struct that holds a memory map of a `repodata.json` file and also a
+// self-referential field which indexes the data in the memory map with a
+// sparsely parsed json struct. See [`LazyRepoData`].
+#[cfg(any(unix, windows))]
+self_cell::self_cell!(
```

Comment:
I replaced ouroborous with self_cell because I couldnt get to conditionally compile ouroboros. self_cell should also compile faster anyway.